cdap-app-templates/cdap-etl/cdap-etl-api-spark/src/main/java/co/cask/cdap/etl/api/batch/SparkPluginContext.java

```diff
@@ -27,7 +28,12 @@
 
   /**
    * Sets a {@link SparkConf} to be used for the Spark execution.
+   * Ideally {@link PipelineConfigurer#setPipelineProperties} method should be used
```

Comment:
If your configuration will not change between pipeline runs, use PipelineConfigurer#setPipelineProperties instead. This method should only be used when you need different configuration settings for each run.

Due to limitations in Spark Streaming, this method cannot be used in realtime data pipelines. Calling this method will throw an UnsupportedOperationException in realtime pipelines.